templates/launch.html

```diff
@@ -460,6 +463,58 @@ <h3>Runner Data</h3>
         document.getElementById('schema-url-survey-type').selectedIndex = -1;
     }
 
+    function retrieveSchemaName() {
+        const availableSchemas = [...document.querySelector("#schema_name").options].map(x => x.value)
+        let schemaName = localStorage.getItem("schema_name");
+        if (schemaName && availableSchemas.includes(schemaName)) {
+            document.querySelector("#schema_name").value = schemaName;
+
+            toggleSubmitFlushButtons("show");
+            clearAndDisableSchemaUrlSelection()
+
+            const surveyType = document.querySelector(`#schema_name option[value="${schemaName}"]`).dataset.surveyType
+
+            loadSurveyMetadata(schemaName, surveyType)
+            loadSchemaMetadata(schemaName, null)
+            document.querySelector("#local-storage-btn").classList.remove("btn--hidden");
+        }
+        else {
+            document.querySelector("#local-storage-btn").classList.add("btn--hidden");
+        }
+    }
+
+    function retrieveResponseId() {
+        let responseId = localStorage.getItem("response_id");
+
+        if (responseId) {
+            document.querySelector("#response-id-btn").classList.remove("btn--hidden");
```

Comment:
Minor, avoid the duplication where possible, multiple patterns of reusing the same thing more than once. Use a variable so the element so `let responseIdButton = document.querySelector("#response-id-btn")` etc.